components/centraldashboard-angular/frontend/src/app/pages/main-page/main-page.component.ts

```diff
@@ -63,4 +83,45 @@ export class MainPageComponent implements OnInit, OnDestroy {
   computeBuildValue(label: string, buildValue: string): string {
     return `${label} ${buildValue}`;
   }
+
+  handleDashboardLinks(links: DashboardLinks) {
+    const { menuLinks, externalLinks, quickLinks, documentationItems } = links;
+    this.menuLinks = menuLinks || [];
+    this.externalLinks = externalLinks || [];
+    this.quickLinks = quickLinks || [];
+    this.documentationItems = documentationItems || [];
+  }
+
+  getUrlPath(url: string) {
+    const urlWithoutFragment = url.split('#')[0];
+    return this.appendPrefix(urlWithoutFragment);
+  }
+
+  getUrlFragment(url: string): string {
+    const fragment = url.split('#')[1];
+    return fragment;
+  }
+
+  appendPrefix(url: string): string {
+    return '/_' + url;
+  }
+
+  isLinkActive(url: string): boolean {
+    let browserUrl = this.router.url;
+    browserUrl = appendBackslash(browserUrl);
+    url = appendBackslash(url);
+    /**
+     * We need this condition because Pipelines Web App removes
+     * the 's' from its path when navigating inside a Recurring
+     * Run's details page
+     */
+    if (
+      browserUrl.includes('pipeline/#/recurringrun') &&
+      url.includes('/pipeline/#/recurringruns')
+    ) {
+      return true;
+    }
+
+    return browserUrl.includes(url);
```

Comment:
It works, but it will be safer to use `startsWith()` in case we end up with a `url` inside the `browserUrl` that we don't want to match

Reply:
Sure, I removed the prefix from browser's URL and changed that to `startsWith()`.